packages/angular/cli/models/schematic-command.ts

```diff
@@ -242,14 +242,17 @@ export abstract class SchematicCommand<
       schemaValidation: true,
       optionTransforms: [
         // Add configuration file defaults
-        async (schematic, current) => ({
-          ...(await getSchematicDefaults(
-            schematic.collection.name,
-            schematic.name,
-            getProjectName(),
-          )),
-          ...current,
-        }),
+        async (schematic, current) => {
+          const projectName =
+            typeof (current as Record<string, unknown>).project === 'string'
+              ? ((current as Record<string, unknown>).project as string)
+              : getProjectName();
+
+          return {
```

Comment:
We cannot validate the project name at this stage, since the project might now exist yet, example when using `ng g lib` or `ng g app`, when under the will invoke `ng g c` will an unknown project name.